src/LazyOperations/CartesianProduct.jl

```diff
@@ -296,22 +296,21 @@ function concretize(cp::CartesianProduct)
 end
 
 function project(cp::CartesianProduct, block::AbstractVector{Int})
-    res = _project_cp_same_block(cp, block)
-    if res == nothing
-        res = _project_linear_map(cp, block)
-    end
-    return res
-end
-
-function _project_cp_same_block(cp, block)
-    min, max = extrema(block)
     n1 = dim(cp.X)
-    if max <= n1
+    if block[end] <= n1
```

Comment:
i dont know if we have here or in a more global description of the docs that `block` should be increasing integers.

Reply:
True. In many other places we use `v[block]` for vector `v`, which gives wrong results if you break this assumption. For example:

```julia
julia> project(Singleton([1, 2, 3]), [3, 1])
Singleton{Int64,Array{Int64,1}}([3, 1])
```